src/Register/index.test.js

```diff
@@ -1,91 +1,136 @@
-          password: 'password',
-          id: 1,
-        },
-      ],
+const mocks = [
+  {
+    request: { query: QUERY_USERS },
+    result: {
+      data: {
+        users: [
+          {
+            username: 'User',
+            password: 'password',
+            id: 1,
+          },
+        ],
+      },
     },
   },
-};
-
-const mockedRegisterResponse = {
-  request: {
-    query: REGISTER,
-    variables: {
-      username: 'User_',
-      firstName: 'John',
-      lastName: 'Smith',
-      password: 'password',
+  {
+    request: {
+      query: REGISTER,
+      variables: {
+        username: 'User_',
+        firstName: 'John',
+        lastName: 'Smith',
+        password: 'password',
+      },
     },
-  },
-  result: {
-    data: {
-      id: 1,
+    result: {
+      data: {
+        id: 1,
+      },
     },
   },
-};
-
-const mocks = [
-  mockedUsersResponse,
-  mockedUsersResponse,
-  mockedRegisterResponse,
 ];
 
-it('renders the register page', async () => {
-  render(
-    <MockedProvider mocks={mocks} addTypename={false}>
-      <AuthProvider>
-        <Router>
-          <Register />
-        </Router>
-      </AuthProvider>
-    </MockedProvider>,
-  );
-
-  expect(await screen.findByRole('form')).toBeInTheDocument();
+describe('Register page', () => {
+  const setup = () => {
+    render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <AuthProvider>
+          <Router initialEntries={[REGISTER_PATH]}>
+            <Switch>
+              <Route
+                exact
+                path={HOME_PATH}
+                component={() => <div>Home</div>}
+              />
+              <Route path={REGISTER_PATH} component={Register} />
+            </Switch>
+          </Router>
+        </AuthProvider>
+      </MockedProvider>,
+    );
+    const usernameField = screen.getByLabelText('Username');
+    const firstNameField = screen.getByLabelText('First Name');
+    const lastNameField = screen.getByLabelText('Last Name');
+    const passwordField = screen.getByLabelText('Password');
+    const registerButton = screen.getByRole('button');
+    return {
+      usernameField, firstNameField, lastNameField, passwordField, registerButton,
+    };
+  };
 
-  const usernameField = screen.getByLabelText('Username');
-  userEvent.type(usernameField, ' {backspace}');
-  expect(await screen.findByText('Username required.')).toBeInTheDocument();
-  userEvent.type(usernameField, 'User');
-  expect(screen.getByDisplayValue('User')).toBeInTheDocument();
+  it('renders the register page', () => {
+    const {
+      usernameField, firstNameField, lastNameField, passwordField, registerButton,
+    } = setup();
+    expect(screen.getByRole('form')).toBeInTheDocument();
+    expect(usernameField).toBeInTheDocument();
+    expect(firstNameField).toBeInTheDocument();
+    expect(lastNameField).toBeInTheDocument();
+    expect(passwordField).toBeInTheDocument();
+    expect(registerButton).toBeInTheDocument();
```

Comment:
These assertions are redundant since we already do the same thing inside `setup()` with `getBy*`:
```
    expect(usernameField).toBeInTheDocument();
    expect(firstNameField).toBeInTheDocument();
    expect(lastNameField).toBeInTheDocument();
    expect(passwordField).toBeInTheDocument();
    expect(registerButton).toBeInTheDocument();
```